commit e5375f64064efc1d18a3809dc12d1ea1376dcb45
Author: Kalesh Singh <kaleshsingh@google.com>
Date:   Sun Oct 13 11:56:16 2024 -0700

    ltp: syscalls: signal: Fix alignment and size of altstack
    
    On architectures such as arm64 the base page size can be
    as large as 64KiB.
    
    Introduce a MAX_PAGE_SIZE and use this for compile time
    page aligment and sizes.
    
    Bug: 366098040
    Test: atest vts_ltp_test_x86_64:syscalls.signal06_64bit#syscalls.signal06_64bit
    Change-Id: I86e3f4d3574bbe86020bfc71cee845f97041d63a
    Signed-off-by: Kalesh Singh <kaleshsingh@google.com>

diff --git a/include/pgsize_helpers.h b/include/pgsize_helpers.h
index 7e57a7456..9c613a5bc 100644
--- a/include/pgsize_helpers.h
+++ b/include/pgsize_helpers.h
@@ -9,6 +9,8 @@
 
 #include <unistd.h>
 
+#define MAX_PAGE_SIZE (64*1024)
+
 #if defined(__x86_64__)
 /*
  * Android emulates the userspace page size on some x86_64 emulators.
diff --git a/testcases/kernel/syscalls/signal/signal06.c b/testcases/kernel/syscalls/signal/signal06.c
index cf0706ea4..1a28b2767 100644
--- a/testcases/kernel/syscalls/signal/signal06.c
+++ b/testcases/kernel/syscalls/signal/signal06.c
@@ -49,6 +49,7 @@
 
 #include "test.h"
 #include "lapi/syscalls.h"
+#include "pgsize_helpers.h"
 
 char *TCID = "signal06";
 int TST_TOTAL = 5;
@@ -61,7 +62,7 @@ int TST_TOTAL = 5;
 volatile double D;
 volatile int FLAGE;
 
-char altstack[4096 * 10] __attribute__((aligned(4096)));
+char altstack[MAX_PAGE_SIZE * 10] __attribute__((aligned(MAX_PAGE_SIZE)));
 
 void test(void)
 {
